### ISSUE TITLE
Rule to enforce import module only

### ISSUE THREAD

**Issue by justinchuby** [charliermarsh/ruff#3045]
The google style guide recommends importing modules only. https://google.github.io/styleguide/pyguide.html#22-imports. It would be great if there’s a rule to enforce that.

**Comment by Avasam** [comment]
If I understand correctly, you'd want https://docs.astral.sh/ruff/rules/banned-import-from/ but for all imports ? (which could be done if globbing is supported, see https://github.com/astral-sh/ruff/issues/2660 )

**Comment by justinchuby** [comment]
Not exactly. From import should still be allowed if the imported name is a module. 

E.g.

from torch import n

should be allowed because nn is a module.

**Comment by qthequartermasterman** [comment]
https://github.com/Enforcer/pylint_google_style_guide_imports_enforcing exists as a reference implementation. It is a pylint plugin.

**Comment by revmischa** [comment]
I opened a PR to enforce the google style guide rules #21118

**Comment by qthequartermasterman** [comment]
It's tragic the PR was closed. The [pylint plugin](https://github.com/Enforcer/pylint_google_style_guide_imports_enforcing) enforcing this feature is the only things keeping us on pylint still.

**Comment by serjflint** [comment]
@ntBre Hello! Is it possible to contribute this feature as on opt-in option to isort section of the formatter instead of a separate rule for the linter?

**Comment by ntBre** [comment]
Hi @serjflint! Thanks for checking! But this doesn't seem like it has much overlap with isort to me. This is about linting against particular types of imports, not about grouping or sorting them in a certain way, unless I'm misunderstanding something.

**Comment by serjflint** [comment]
> Hi @serjflint! Thanks for checking! But this doesn't seem like it has much overlap with isort to me. This is about linting against particular types of imports, not about grouping or sorting them in a certain way, unless I'm misunderstanding something.

I meant forcing imports to import only modules. But knowing which is a module or an object inside a module is a bit outside of scope of the formatter.

**Comment by paveldikov** [comment]
Is it possible to reconsider the closure of the previous PR? It seemed like a very simple implementation, and it being gated behind preview made sense.

Lots of enterprise 'Python house styles' (not just Google's!) recommend/mandate module imports (reduction of naming tautologies and namespace pollution being some of the key benefits).

At least in my world, currently we have to police this by hand -- or hope that AI assistants/reviewers will respect the rule. Obviously that is rather futile. I'd much rather rely on the linter instead! :-)
